### ISSUE TITLE
PHP warning: inet_pton(): Unrecognized address unknow

### ISSUE THREAD

**Issue by rwetzlmayr** [SocialEngine/phpv4-issues#330]
`\Engine_IP::getRealRemoteAddress` happily accepts anything as the client's IP address from the `$_SERVER` global.  

Malformed or malicious client requests thus can inject arbitrary strings into it, and as real life shows, they do supply strings like 'unknown' which causes the PHP warning mentioned in this issue's title.

I propose to be less liberal. Here's a patch:

```diff
--- a/web/application/libraries/Engine/IP.php
+++ b/web/application/libraries/Engine/IP.php
@@ -320,10 +320,12 @@
       $asIPv6 = self::$_preferIPv6;
     }
 
+    $address = filter_var($address, FILTER_VALIDATE_IP);
+
     if( $asIPv6 ) {
       return self::convertIPv4to6($address);
     } else {
       return $address;
     }
   }
   
```

**Comment by smazurov** [comment]
Thanks for the patch, we'll add this in next release!

**Comment by smazurov** [comment]
Fixed in [4.8.12](http://blog.socialengine.com/2016/07/07/socialengine-php-4-8-12-is-here/)
